R/lazyframe__lazy.R

```diff
@@ -1397,8 +1404,108 @@ LazyFrame_fetch = function(
 #'   group_by("Species", maintain_order = TRUE)$
 #'   agg(pl$col(pl$Float64)$apply(r_func))$
 #'   profile()
-LazyFrame_profile = function() {
-  .pr$LazyFrame$profile(self) |> unwrap("in $profile()")
+LazyFrame_profile = function(
+  type_coercion = TRUE,
+  predicate_pushdown = TRUE,
+  projection_pushdown = TRUE,
+  simplify_expression = TRUE,
+  slice_pushdown = TRUE,
+  comm_subplan_elim = TRUE,
+  comm_subexpr_elim = TRUE,
+  streaming = FALSE,
+  no_optimization = FALSE,
+  inherit_optimization = FALSE,
+  collect_in_background = FALSE,
+  show_plot = FALSE,
+  truncate_nodes = 0) {
+
+  if (isTRUE(no_optimization)) {
+    predicate_pushdown = FALSE
+    projection_pushdown = FALSE
+    slice_pushdown = FALSE
+    comm_subplan_elim = FALSE
+    comm_subexpr_elim = FALSE
+  }
+
+  if (isTRUE(streaming)) {
+    comm_subplan_elim = FALSE
+  }
+
+  lf = self
+
+  if (isFALSE(inherit_optimization)) {
+    lf = self$set_optimization_toggle(
+      type_coercion,
+      predicate_pushdown,
+      projection_pushdown,
+      simplify_expression,
+      slice_pushdown,
+      comm_subplan_elim,
+      comm_subexpr_elim,
+      streaming
+    ) |> unwrap("in $profile():")
+  }
+
+  out = lf |>
+    .pr$LazyFrame$profile() |>
+    unwrap("in $profile()")
+
+  if (isTRUE(show_plot)) {
+    if (!requireNamespace("ggplot2", quietly = TRUE)) {
+      stop('The package "ggplot2" is required.')
+    }
+    timings = out$profile$to_data_frame()
+    timings$node = factor(timings$node, levels = unique(timings$node))
+    total_timing = max(timings$end)
+    if (total_timing > 10000000) {
+      unit = "s"
+      total_timing = paste0(total_timing/1000000, "s")
+      timings$start = timings$start / 1000000
+      timings$end = timings$end / 1000000
+    } else if (total_timing > 10000) {
+      unit = "ms"
+      total_timing = paste0(total_timing/1000, "ms")
+      timings$start = timings$start / 1000
+      timings$end = timings$end / 1000
+    } else {
+      unit = "\U00B5s"
+      total_timing = paste0(total_timing, "\U00B5s")
+    }
+
+    # for some reason, there's an error if I use rlang::.data directly in aes()
+    .data = rlang::.data
```

Comment:
If using rlang directly I guess it should be a soft dependency and/or check for it is available together with ggplot2 right?

Reply:
`rlang` is in Suggests and we don't need to check here because  if `ggplot2` is installed then `rlang` is installed too